src/units/abilities.cpp

```diff
@@ -867,41 +867,118 @@ static void add_name(std::string& weapon_abilities, bool active, const config::a
 std::string attack_type::weapon_specials(bool only_active, bool is_backstab) const
 {
 	//log_scope("weapon_specials");
-	std::string res;
-	for (const config::any_child sp : specials_.all_children_range())
-	{
-		const bool active = special_active(sp.cfg, AFFECT_EITHER, sp.key, is_backstab);
+	return weapon_specials_impl(only_active, is_backstab);
+}
 
-		const std::string& name =
-			active
-			? sp.cfg["name"].str()
-			: default_value(sp.cfg, "name_inactive", "name").str();
-		if (!name.empty()) {
-			if (!res.empty()) res += ", ";
-			if (only_active && !active) res += font::span_color(font::inactive_details_color);
-			res += name;
-			if (only_active && !active) res += "</span>";
+std::string attack_type::weapon_specials_value(const std::set<std::string> checking_tags, bool only_active, bool is_backstab) const
+{
+	//log_scope("weapon_specials_value");
+	return weapon_specials_impl(only_active, is_backstab, checking_tags, true);
+}
+
+std::string attack_type::weapon_specials_impl(bool only_active, bool is_backstab, const std::set<std::string> checking_tags, bool is_tooltip) const
+{
+	std::string res;
+	for (const config::any_child sp : specials_.all_children_range()) {
+		bool tag_checked = is_tooltip ? (checking_tags.count(sp.key) != 0) : true;
+		if(tag_checked){
+			const bool active = is_tooltip ? special_active(sp.cfg, AFFECT_SELF, sp.key, is_backstab) : special_active(sp.cfg, AFFECT_EITHER, sp.key, is_backstab);
```

Comment:
On this one, I'm not understanding why the constructing a tooltip would use different values.

Reply:
In tooltip, i want what only weapon_special who affect directly stats in tooltip was showed, if applyto=opponent, name appear in attacker in windows because owner of special, but in defender tooltip only because defender stats modified by special, and AFFECT_SELF used instead of AFFECT_EITHER.

Reply:
and for name string, if inactive_name can appear in window, what inactive name appear in attacker tooltip then what special active but applied to defender is confusing, and name must't appear in that case.